### PR TITLE
Acceptance test for detecting unsupported v2 app.config configuration

### PR DESCRIPTION
**Pull request by tmasternak**

### CODE REVIEW

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    public abstract class AppConfig : IDisposable
+    {
+        public static AppConfig Change(string path)
+        {
+            return new ChangeAppConfig(path);
+        }
+
+        public abstract void Dispose();
+
+        private class ChangeAppConfig : AppConfig
```
Comment (andreasohlund) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs]: Redundant private

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                {
+                    AppDomain.CurrentDomain.SetData("APP_CONFIG_FILE", oldConfig);
+                    ResetConfigMechanism();
+
+                    disposedValue = true;
+                }
+                GC.SuppressFinalize(this);
+            }
+
+            private static void ResetConfigMechanism()
```
Comment (andreasohlund) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs]: Redundant private

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        public static AppConfig Change(string path)
+        {
+            return new ChangeAppConfig(path);
+        }
+
+        public abstract void Dispose();
+
+        private class ChangeAppConfig : AppConfig
+        {
+            private readonly string oldConfig =
```
Comment (andreasohlund) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs]: Redundant private

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                AppDomain.CurrentDomain.GetData("APP_CONFIG_FILE").ToString();
+
+            private bool disposedValue;
```
Comment (andreasohlund) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs]: Redundant private

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/When_using_v2_configuration_app_config.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_using_v2_configuration_app_config : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_fail_on_startup()
+        {
+            var appConfigFilename = $"app.sqlv2.config";
+            var expectedErrorMessage = "Schema override in connection string is not supported anymore";
```
Comment (andreasohlund) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/When_using_v2_configuration_app_config.cs]: Is it only the schema that shouldn't be used? (update the test name in that case?)

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/When_using_v2_configuration_app_config.cs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        }
+
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.UseTransport<SqlServerTransport>();
```
Comment (andreasohlund) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/When_using_v2_configuration_app_config.cs]: Is this needed? (won't the tests default to sqlserver)

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs (hunk 1 above)
Comment (tmasternak) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs]: :+1:

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs (hunk 3 above)
Comment (tmasternak) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs]: :+1:

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs (hunk 4 above)
Comment (tmasternak) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs]: :+1:

File: src/NServiceBus.SqlServer/Configuration/ConnectionStringsValidator.cs [hunk 7]
```diff
@@ -11,6 +11,14 @@ class ConnectionStringsValidator
         const string TransportConnectionStringPrefix = "NServiceBus/Transport";
         const string SchemaOverridePart = "Queue Schema";
 
+        const string MultiDatabaseNotSupported =
```
Comment (andreasohlund) [inline src/NServiceBus.SqlServer/Configuration/ConnectionStringsValidator.cs]: Do we have any tests for checking this?

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs (hunk 2 above)
Comment (tmasternak) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/AppConfig.cs]: :+1:

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/When_using_v2_configuration_app_config.cs (hunk 5 above)
Comment (tmasternak) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/When_using_v2_configuration_app_config.cs]: Yes the Queue Schema is unsupported parameter. I've updated the test name.

File: src/NServiceBus.SqlServer/Configuration/ConnectionStringsValidator.cs (hunk 7 above)
Comment (tmasternak) [inline src/NServiceBus.SqlServer/Configuration/ConnectionStringsValidator.cs]: We have all/most of the v2 configuration cases covered [here](https://github.com/Particular/NServiceBus.SqlServer/blob/develop/src/NServiceBus.SqlServer.UnitTests/ConfigurationValidatorTests.cs). This test is to make sure that the validation does happen when the endpoint starts.

Comment (andreasohlund) [inline src/NServiceBus.SqlServer/Configuration/ConnectionStringsValidator.cs]: :+1:

File: src/NServiceBus.SqlServer.AcceptanceTests/Configuration/When_using_v2_configuration_app_config.cs (hunk 6 above)
Comment (tmasternak) [inline src/NServiceBus.SqlServer.AcceptanceTests/Configuration/When_using_v2_configuration_app_config.cs]: :+1:

**Pull request merged**